bokeh/models/annotations.py

```diff
@@ -111,8 +112,9 @@ class BoxAnnotation(Annotation):
 
     """
 
-    left = Either(Auto, NumberSpec(), default=None, help="""
-    The x-coordinates of the left edge of the box annotation.
+    left = NumberSpec(default=None, help="""
+    The x-coordinates of the left edge of the box annotation. If value is set
+    to `None`, the box annotation will continue to the edge of the plot area.
     """)
```

Comment:
So i'm not sure the original code ever worked as intended. The dimension parameters didn't work as specs (there wasn't any sort of datasource lookup) and there was no JS-side code to handle any sort of "auto" values.

The annotation examples in the user guide also still work (using default `None` value and not Auto) the same with the new implementation.

Do you want to add Auto and do the transforms or just have the dimensions be NumberSpecs?